docs/content/drafts/UnderlineNav2.mdx

```diff
@@ -1,38 +1,129 @@
   <UnderlineNav.Item selected>Item 1</UnderlineNav.Item>
   <UnderlineNav.Item>Item 2</UnderlineNav.Item>
   <UnderlineNav.Item>Item 3</UnderlineNav.Item>
 </UnderlineNav>
 ```
 
+### Small variant
+
+```jsx live drafts
+<UnderlineNav variant="small">
+  <UnderlineNav.Item selected>Item 1</UnderlineNav.Item>
+  <UnderlineNav.Item>Item 2</UnderlineNav.Item>
+</UnderlineNav>
+```
+
 ### With icons
 
 ```jsx live drafts
-<UnderlineNav label="simple nav with icons">
-  <UnderlineNav.Item selected leadingIcon={EyeIcon}>
-    Item 1
+<UnderlineNav>
+  <UnderlineNav.Item selected leadingIcon={CodeIcon}>
+    Code
+  </UnderlineNav.Item>
+  <UnderlineNav.Item leadingIcon={IssueOpenedIcon} counter={30}>
+    Issues
+  </UnderlineNav.Item>
+  <UnderlineNav.Item leadingIcon={GitPullRequestIcon} counter={3}>
+    Pull requests
+  </UnderlineNav.Item>
+  <UnderlineNav.Item leadingIcon={CommentDiscussionIcon}>Discussions</UnderlineNav.Item>
+  <UnderlineNav.Item leadingIcon={EyeIcon} counter={9}>
+    Actions
+  </UnderlineNav.Item>
+  <UnderlineNav.Item leadingIcon={EyeIcon} counter={7}>
+    Projects
   </UnderlineNav.Item>
-  <UnderlineNav.Item>Item 2</UnderlineNav.Item>
 </UnderlineNav>
 ```
 
-### Small variant
+### Overflow Behaviour
+
+#### Items without Icons
+
+When overflow occurs, the component first hides icons if present to optimize for space and show as many items as possible.
+If there is still overflow, the component will behave depending on the pointer. Please see below to see these behaviours.
 
 ```jsx live drafts
-<UnderlineNav label="small variant" variant="small">
-  <UnderlineNav.Item selected>Item 1</UnderlineNav.Item>
-  <UnderlineNav.Item>Item 2</UnderlineNav.Item>
+<UnderlineNav>
+  <UnderlineNav.Item selected leadingIcon={CodeIcon}>
+    Code
+  </UnderlineNav.Item>
+  <UnderlineNav.Item leadingIcon={IssueOpenedIcon} counter={30}>
+    Issues
+  </UnderlineNav.Item>
+  <UnderlineNav.Item leadingIcon={GitPullRequestIcon} counter={3}>
+    Pull requests
+  </UnderlineNav.Item>
+  <UnderlineNav.Item leadingIcon={CommentDiscussionIcon}>Discussions</UnderlineNav.Item>
+  <UnderlineNav.Item leadingIcon={PlayIcon} counter={9}>
+    Actions
+  </UnderlineNav.Item>
+  <UnderlineNav.Item leadingIcon={ProjectIcon} counter={7}>
+    Projects
+  </UnderlineNav.Item>
+  <UnderlineNav.Item leadingIcon={{navigation: 'Security', icon: ShieldLockIcon}}>Security</UnderlineNav.Item>
+  <UnderlineNav.Item leadingIcon={GraphIcon}>Insights</UnderlineNav.Item>
+  <UnderlineNav.Item leadingIcon={GearIcon} counter={1}>
+    Settings
+  </UnderlineNav.Item>
+</UnderlineNav>
+```
+
+#### Display `More` menu for fine pointer devices
+
+This overflow behaviour will occur on fine pointer devices that have fine pointers.
+
+#### Scroll for coarse pointer devices
+
+The scroll behaviour will occur on devices that have coarse pointers.
+
+Depending on your device when viweing this example, you may see the scroll behaviour or the `More` menu behaviour.
```

Comment:
Help needed:  I am not sure if we can simulate coarse pointers as another example here?